File: 0198.HouseRobber.go
func rob(nums []int) int {
    pre, cur := 0, 0
    for _, num := range nums {
        pre, cur = cur, max(pre+num, cur)
    }
    return cur
}

func max(x, y int) int {
    if x > y {
        return x
    }
    return y
}

File: 0000-0999/0198.HouseRobber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        '''
        n = len(nums) + 2
        dp = [0] * n
        for i in xrange(2, n):
            dp[i] = max(dp[i-1], dp[i-2]+nums[i-2])
        return dp[-1]
        '''
        last, now = 0, 0
        for num in nums:
            last, now = now, max(now, last+num)
        return now